### PR TITLE
Add cancellation token to react to Copilot API changes

### PR DESCRIPTION
**Pull request by benmcmorran**
Reacts to upstream changes in the Copilot API that add a cancellation token and make it possible to unregister a provider.

FYI @sandersn

### CODE REVIEW

**Review by sean-mcmanus — DISMISSED** [review]

**Comment by sean-mcmanus** [comment]
@benmcmorran Are you not able to squash and merge? I could try switching your permissions from Write to Maintain.

File: Extension/src/LanguageServer/extension.ts [hunk 1]
```diff
@@ -1403,7 +1403,9 @@ export async function preReleaseCheck(): Promise<void> {
 }
 
 export async function getIncludes(maxDepth: number): Promise<any> {
-    const includes = await clients.ActiveClient.getIncludes(maxDepth);
+    const tokenSource = new vscode.CancellationTokenSource();
+    const includes = await clients.ActiveClient.getIncludes(maxDepth, tokenSource.token);
```
Comment (Colengms) [inline Extension/src/LanguageServer/extension.ts:1407]: We'd 'fixed' LSP cancelation such that it triggers an exception on cancellation.  However, it doesn't trigger the exception that VS Code generally expects, which would be a `CancellationError`.  I'm not sure what the caller would expect in this case.  Do you know?

**Review by Colengms — CHANGES_REQUESTED** [review]
(See existing comment.  Waiting for clarification before approving.)

File: Extension/src/LanguageServer/extension.ts (hunk 1 above)
Comment (benmcmorran) [reply Extension/src/LanguageServer/extension.ts:1407]: Thanks, I've checked and updated this to use the same VS Code semantics as the chat context request.

**Comment by benmcmorran** [comment]
@Colengms I've changed the cancellation handling to follow the same pattern used by chat context. Let me know if you still have any concerns.

**Review by Colengms — APPROVED** [review]

**Comment by benmcmorran** [comment]
@sean-mcmanus I'm still not able to squash and merge. I suspect there are extra branch polices on `main` specifically that don't include me.

image

**Comment by sean-mcmanus** [comment]
> @sean-mcmanus I'm still not able to squash and merge. I suspect there are extra branch polices on `main` specifically that don't include me.

You should have permissions now.

**Review by sean-mcmanus — APPROVED** [review]

**Pull request merged**